src/Servers/Kestrel/shared/KnownHeaders.cs

```diff
@@ -94,7 +94,11 @@ static KnownHeaders()
                 "Translate",
                 "User-Agent",
                 "DNT",
-                "Upgrade-Insecure-Requests"
+                "Upgrade-Insecure-Requests",
+                "Request-Id",
+                "Correlation-Context",
+                "TraceParent",
+                "TraceState"
```

Comment:
So I see you *did* come up with a few more non-HTTP/2 headers. Do these make sense to you @Tratcher?

Reply:
They are from hosting as it was checking each header that was the same length with IgnoreOrdinalCase for
```csharp
if (!headers.TryGetValue(TraceParentHeaderName, out var correlationId))
{
    headers.TryGetValue(RequestIdHeaderName, out correlationId);
```